lib/alexa_verifier/verifier.rb

```diff
@@ -117,7 +117,7 @@ def check_that_request_was_signed(certificate_public_key, request, raw_body)
         OpenSSL::Digest::SHA1.new,
         Base64.decode64(request.env['HTTP_SIGNATURE']),
         raw_body
-      )
+      ) if !request.env['HTTP_SIGNATURE'].nil? && !request.env['HTTP_SIGNATURE'].empty?
```

Comment:
I might suggest verifying that these fields are present before this line, and throwing a more specific error in the case that they're not.  This is, for example, how we check if the time field is present above:

https://github.com/sidoh/alexa_verifier/blob/f3834d27817818272e1b884e29293570a11911f5/lib/alexa_verifier/verifier.rb#L83

Should still raise an `InvalidRequestError`, but can make the message something like 

> `HTTP_SIGNATURE` header is required, but was not present.